concrete/src/File/ImportProcessor/AutorotateImageProcessor.php

```diff
@@ -0,0 +1,32 @@
+<?php
+namespace Concrete\Core\File\ImportProcessor;
+
+use Concrete\Core\Entity\File\Version;
+use Concrete\Core\File\Type\Type as FileType;
+use Concrete\Core\Support\Facade\Image;
+use Imagine\Filter\Basic\Autorotate;
+use Imagine\Filter\Transformation;
+use Imagine\Image\Metadata\ExifMetadataReader;
+
+class AutorotateImageProcessor implements ProcessorInterface
+{
+    public function shouldProcess(Version $version)
+    {
+        return function_exists('exif_read_data')
+                && $version->getTypeObject()->getGenericType() == FileType::T_IMAGE;
```

Comment:
wouldn't it be better to check for JPG's here? PNG and GIF don't contain exif information. I didn't check what happens right now, but it might be faster anyway.